libs/community/langchain_google_community/gcs_file.py

```diff
@@ -0,0 +1,93 @@
+import os
+import tempfile
+from typing import Callable, List, Optional
+
+from langchain_core.document_loaders import BaseLoader
+from langchain_core.documents import Document
+
+from langchain_google_community._utils import get_client_info
+
+
+class GCSFileLoader(BaseLoader):
+    """Load from GCS file."""
+
+    def __init__(
+        self,
+        project_name: str,
+        bucket: str,
+        blob: str,
+        loader_func: Optional[Callable[[str], BaseLoader]] = None,
+    ):
+        """Initialize with bucket and key name.
+
+        Args:
+            project_name: The name of the project to load
+            bucket: The name of the GCS bucket.
+            blob: The name of the GCS blob to load.
+            loader_func: A loader function that instantiates a loader based on a
+                file_path argument. If nothing is provided, the
+                UnstructuredFileLoader is used.
+
+        Examples:
+            To use an alternative PDF loader:
+            >> from from langchain_community.document_loaders import PyPDFLoader
+            >> loader = GCSFileLoader(..., loader_func=PyPDFLoader)
+
+            To use UnstructuredFileLoader with additional arguments:
+            >> loader = GCSFileLoader(...,
+            >>      loader_func=lambda x: UnstructuredFileLoader(x, mode="elements"))
+
+        """
+        self.bucket = bucket
+        self.blob = blob
+        self.project_name = project_name
+
+        def default_loader_func(file_path: str) -> BaseLoader:
+            try:
+                from langchain_community.document_loaders.unstructured import (
+                    UnstructuredFileLoader,
+                )
+            except ImportError:
+                message = (
+                    "UnstructuredFileLoader loader not found! Either provide a "
+                    "custom loader with loader_func argument, or install "
+                    " `pip install langchain_community`"
+                )
+                print(message)
+            return UnstructuredFileLoader(file_path)
+
+        self._loader_func = loader_func if loader_func else default_loader_func
+
+    def load(self) -> List[Document]:
+        """Load documents."""
+        try:
+            from google.cloud import storage  # type: ignore[attr-defined]
+        except ImportError:
+            raise ImportError(
+                "Could not import google-cloud-storage python package. "
+                "Please install it with `pip install google-cloud-storage`."
+            )
```

Comment:
Why not adding this as poetry dependencies instead of loading dependencies this way?

Reply:
that might be the next step, but we need to create separate groups.
we don't want to bring all dependencies together as a requirements for the package.